Parse/ParseClientConfiguration.h

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Parse/PFConstants.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol ParseMutableClientConfiguration<NSObject>
+
+@property (nullable, nonatomic, copy, readwrite) NSString *applicationId;
+@property (nullable, nonatomic, copy, readwrite) NSString *clientKey;
+
+@property (nonatomic, assign, readwrite, getter=isLocalDatastoreEnabled) BOOL localDatastoreEnabled;
+
+@property (nullable, nonatomic, copy, readwrite) NSString *applicationGroupIdentifier;
+@property (nullable, nonatomic, copy, readwrite) NSString *containingApplicationBundleIdentifier;
+
+@property (null_resettable, nonatomic, copy, readwrite) NSURLSessionConfiguration *URLSessionConfiguration;
```

Comment:
Should we be publicizing this? I'm afraid of the possibility that we change our networking layer at some point in the future and this configuration becomes obsolete.

What kind of configuration does this API provide? Maybe we should abstract these configurations.